storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -347,17 +353,40 @@ func (s *peersSource) incrementalFlush(
 		shardRetriever    = shardRetrieverMgr.ShardRetriever(shard)
 		tmpCtx            = context.NewContext()
 		seriesCachePolicy = s.opts.ResultOptions().SeriesCachePolicy()
+		filePathPrefix    = s.opts.FilesystemOptions().FilesystemOptions().FilePathPrefix()
 	)
 	if seriesCachePolicy == series.CacheAllMetadata && shardRetriever == nil {
 		return fmt.Errorf("shard retriever missing for shard: %d", shard)
 	}
 
 	for start := tr.Start; start.Before(tr.End); start = start.Add(blockSize) {
-		prepared, err := flush.Prepare(nsMetadata, shard, start)
+		prepared, ok, err := flush.Prepare(nsMetadata, shard, start)
 		if err != nil {
 			return err
 		}
 
+		if !ok {
```

Comment:
would be to make this explicit: either have `flush.Prepare` return a typed error (`ErrFilesetAlreadyExists`), or  check `fs.FilesetExistsAt` here

Reply:
I think this is really explicit, the ok boolean does nothing but tell you whether the files already exist or not. Returning a typed error seems worse to me because then all callers have to know to handle that specific type of error, whereas with the boolean you're force to handle that situation because its part of the type signature